tidy3d/plugins/dispersion/fit_fast.py

```diff
@@ -0,0 +1,834 @@
+
+    @classmethod
+    def initialize(
+        cls,
+        omega: ArrayFloat1D,
+        eps: ArrayComplex1D,
+        eps_inf: float,
+        advanced_param: AdvancedFastFitterParam,
+    ) -> FastFitterData:
+        """Construct FastFitterData from AdvancedFastFitterParam"""
+        weights = advanced_param.weights or cls.get_default_weights(eps)
+        optimize_eps_inf = None if eps_inf is None else False
+        data = FastFitterData(
+            min_num_poles=advanced_param.min_num_poles,
+            num_iters=advanced_param.num_iters,
+            passivity_num_iters=advanced_param.passivity_num_iters,
+            slsqp_constraint_scale=advanced_param.slsqp_constraint_scale,
+            loss_bounds=advanced_param.loss_bounds,
+            weights=weights,
+            verbose=advanced_param.verbose,
+            relaxed=advanced_param.relaxed,
+            smooth=advanced_param.smooth,
+            logspacing=advanced_param.logspacing,
+            omega=omega,
+            eps=eps,
+            optimize_eps_inf=optimize_eps_inf,
+            eps_inf=eps_inf or 1,
+        )
+        return data
+
+    @cached_property
+    def real_poles(self) -> ArrayFloat1D:
+        """The real poles."""
+        return self.poles[np.isreal(self.poles)]
+
+    @cached_property
+    def complex_poles(self) -> ArrayFloat1D:
+        """The complex poles."""
+        return self.poles[~np.isreal(self.poles)]
+
+    @classmethod
+    def get_default_weights(cls, eps: ArrayComplex1D) -> Tuple[float, float]:
+        """Default weights based on real and imaginary part of eps."""
+        rms = np.array([np.sqrt(np.mean(x**2)) for x in (np.real(eps), np.imag(eps))])
+        rms = np.maximum(RMS_MIN, rms)
+        weights = [1 / val for val in rms]
+        average = (weights[0] + weights[1]) / 2
+        weights = [val / average for val in weights]
+        return tuple(weights)
+
+    @cached_property
+    def pole_residue(self) -> PoleResidue:
+        """Corresponding PoleResidue model"""
+        if self.eps_inf is None or self.poles is None:
+            return None
+        return PoleResidue(
+            eps_inf=self.eps_inf, poles=list(zip(self.poles / HBAR, self.residues / HBAR))
+        )
+
+    def evaluate(self, omega: float) -> complex:
+        """Evaluate model at omega in eV"""
+        eps = self.eps_inf
+        for (pole, res) in zip(self.poles, self.residues):
+            eps += -res / (1j * omega + pole) - np.conj(res) / (1j * omega + np.conj(pole))
+        return eps
+
+    @cached_property
+    def values(self) -> ArrayComplex1D:
+        """Evaluate model at sample frequencies."""
+        return self.evaluate(self.omega)
+
+    @cached_property
+    def imag_extrema(self) -> Tuple[ArrayFloat1D, ArrayComplex1D]:
+        """Extrema of imaginary part of eps."""
+
+        def _extrema_loss_freq_finder(
+            areal, aimag, creal, cimag
+        ):  # pylint: disable=too-many-locals
```

Comment:
just FYI I think it should work to put the disable just above the function signature and might look cleaner.

```py
    # pylint: disable=too-many-locals
    def _extrema_loss_freq_finder(areal, aimag, creal, cimag):
```